File: backend/src/main/java/com/chuangkit/admin/entity/SupportTicket.java
package com.chuangkit.admin.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("support_ticket")
public class SupportTicket {
    @TableId(type = IdType.AUTO) private Long id;
    private Long userId;
    private String subject;
    private String content;
    private String priority;
    private String status;
    private Long assigneeId;
    private String lastReply;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: frontend/src/types/supportTicket.ts
export type SupportTicketStatus = 'open' | 'pending' | 'resolved' | 'closed'
export type SupportTicketPriority = 'low' | 'normal' | 'high' | 'urgent'

export interface SupportTicket {
  id: number
  subject: string
  content: string
  priority: SupportTicketPriority | string
  status: SupportTicketStatus | string
  assigneeId?: number | null
  lastReply?: string | null
  createTime?: string | null
  updateTime?: string | null
}
